app/src/main/java/br/org/cesar/discordtime/stickysessions/domain/repository/UserRepository.java

```diff
@@ -0,0 +1,6 @@
+package br.org.cesar.discordtime.stickysessions.domain.repository;
+
+public interface UserRepository {
```

Comment:
I think it would be better if the UserRepository interface returns Single type from his methods, so you wouldn't need to decide how a particular operation is being executed on UseCases.